src/asmcnc/production/z_head_qc_jig/z_head_qc_2.py

```diff
@@ -289,13 +289,19 @@ def compare_info(dt):
 
     def test_rpm(self, fail_report):
         def read_rpm(dt):
-            if self.m.spindle_speed() < 8000 or self.m.spindle_speed() > 12000:
-                fail_report.append("Spindle RPM was " + str(self.m.spindle_speed()) + ". Should be 8000-12000")
+            spindle_rpm = int(self.m.s.spindle_speed)
+
+            log('Spindle RPM: %s' % spindle_rpm)
+
+            if spindle_rpm < 8000 or spindle_rpm > 12000:
+                fail_report.append("Spindle RPM was " + str(spindle_rpm) + ". Should be 8000-12000")
 
             self.m.s.write_command('M5')
             self.continue_digital_spindle_test(fail_report)
 
-        self.m.s.write_command('M3 S10000')
+        rpm_to_run = 10000 if self.m.spindle_voltage == 230 else self.m.convert_from_110_to_230(10000)
+
+        self.m.s.write_command('M3 S' + str(rpm_to_run))
```

Comment:
For writing commands there's also self.m.send_any_gcode_command() (slightly neater than calling m.s.write_command, but functionally the same :) )